beacon_node/builder_client/src/lib.rs

```diff
@@ -543,9 +535,47 @@ impl BuilderHttpClient {
     }
 }
 
+/// Deserialize a JSON response containing a `ForkVersionedResponse` wrapper.
+///
+/// This handles the case where the response body contains:
+/// ```json
+/// {
+///   "version": "fulu",
+///   "data": { ... }
+/// }
+/// ```
+fn deserialize_json_fork_versioned_response<T>(
```

Comment:
Oh I just noticed I can just move this to implementation for `Deserialize`

Reply:
I think ForkVersionedResponse might already have an impl like this?

Reply:
oh yeah you're right let me fix it